client/lib/login_register/views/login_register_view.dart

```diff
@@ -0,0 +1,259 @@
+import 'package:ccquarters/common_widgets/wide_text_button.dart';
+import 'package:ccquarters/login_register/cubit.dart';
+import 'package:ccquarters/login_register/state.dart';
+import 'package:ccquarters/login_register/views/email_and_password_fields.dart';
+import 'package:ccquarters/login_register/views/personal_info_fields.dart';
+import 'package:ccquarters/model/user.dart';
+import 'package:ccquarters/utils/consts.dart';
+import 'package:ccquarters/utils/device_type.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+
+enum LoginRegisterPageType {
+  login,
+  registerPersonalInfo,
+  registerEmailAndPassword
+}
+
+class LoginRegisterView extends StatefulWidget {
+  const LoginRegisterView({
+    Key? key,
+    required this.page,
+    required this.user,
+  }) : super(key: key);
+
+  final LoginRegisterPageType page;
+  final User user;
+  @override
+  State<LoginRegisterView> createState() => _LoginRegisterViewState();
+}
+
+class _LoginRegisterViewState extends State<LoginRegisterView> {
+  final _formKey = GlobalKey<FormState>();
+  final firstTextField = TextEditingController();
+  final secondTextField = TextEditingController();
+  final thirdTextField = TextEditingController();
+  final fourthTextField = TextEditingController();
+
+  @override
+  void initState() {
+    switch (widget.page) {
+      case LoginRegisterPageType.login:
+        firstTextField.text = widget.user.email;
+        break;
+      case LoginRegisterPageType.registerPersonalInfo:
+        firstTextField.text = widget.user.company ?? "";
```

Comment:
Why not "companyTextField"

Reply:
because i don't use it only to company - also to email in register second page and login page